### ISSUE TITLE
Bounding box cuts of top part of head or shoes

### ISSUE THREAD

**Issue by ridasalam** [dbolya/yolact#88]

**Comment by ridasalam** [comment]
Hi, I've noticed that in some images during evaluation, the bounding box does not fully cover the person and therefore the mask is cut as well. For example, a person standing straight would sometimes have their forehead cut out or part of the shoe at the bottom missing. 

What could be the reason for this?

**Comment by dbolya** [comment]
How often have you seen this happen, and is this on your own dataset?

This is machine learning. The model is bound to make errors like that from time to time, and we can't really explain them.

**Comment by ridasalam** [comment]
Maybe about 20 times in 200 images.
 
Yes, it's on my own dataset. 

Could it be due to the anchor box sizes? The images are pretty simple (plain backgrounds)

**Comment by dbolya** [comment]
Oh then maybe it's just a lack of training? Or it could be overfitting. How many images does your dataset have and how long did you train it (and with what lr settings)?

It could also be that no good anchor box sizes exist, so perhaps. Does your dataset have a lot of one particular aspect ratio / size of box? Maybe you could also tune those settings.

**Comment by ridasalam** [comment]
Another observation I made is that when the object of interest is translated horizontally (same scale), the masks are cut off in some positions.

I have 45k images, trained 36 epochs so far, batchsize=4, lr=5e-04

**Comment by dbolya** [comment]
Hmm those parameters seem fine for a dataset of that size. Maybe you can try to use non-square anchors (there was a bug where all anchors were square). You can enable the fix by turning `'use_square_anchors'` to false in `yolact_base_config`'s `backbone` (or whatever config you're using).

**Comment by ridasalam** [comment]
If I set no_crop in eval, the masks are fine (except some bad background detections in some spaces). Seems like the bb height isn't always good.

**Comment by dbolya** [comment]
Yeah that would be probably be fixed by what I said above, since it sounds like you have a lot of talk boxes? Again, in the current version of the weights, all the anchors are square which would hurt the performance on tall objects.

Worst case I guess you can just manually stretch the bbox height, since you said no crop was working for the most part.

**Comment by ridasalam** [comment]
Sounds good, thanks!
